scripts/run_on_popular_crates/src/main.rs

```diff
@@ -0,0 +1,100 @@
+use std::time::Duration;
+use std::collections::HashMap;
+use std::process::Command;
+use std::io::Write;
+use crates_io_api::{CratesQuery, Sort};
+
+/// To get the rustdoc of the baseline, we first create a placeholder project somewhere
+/// with the baseline as a dependency, and run `cargo rustdoc` on it.
+fn create_rustdoc_manifest_for_crate_version(
+    name: &str,
+    version: &str,
+    features: &HashMap<String, Vec<String>>
+) -> cargo_toml::Manifest<()> {
+    use cargo_toml::*;
+
+    Manifest::<()> {
+        package: {
+            let mut package = Package::new(name, version);
+            package.publish = Inheritable::Set(Publish::Flag(false));
+            Some(package)
+        },
+        workspace: Some(Workspace::<()>::default()),
+        lib: {
+            let product = Product {
+                path: Some("lib.rs".to_string()),
+                ..Product::default()
+            };
+            Some(product)
+        },
+        dependencies: {
+            let project_with_features = DependencyDetail {
+                version: Some(version.to_string()),
+                features: features
+                    .iter()
+                    .map(|(key, _values)| key.clone())
+                    .collect(),
+                ..DependencyDetail::default()
+            };
+            let mut deps = DepsSet::new();
+            deps.insert(
+                name.to_string(),
+                Dependency::Detailed(project_with_features),
+            );
+            deps
+        },
+        ..Default::default()
+    }
+}
+
+fn save_manifest(manifest: cargo_toml::Manifest<()>, path: &str) -> anyhow::Result<()> {
+    std::fs::write(path, toml::to_string(&manifest)?)?;
+    Ok(())
+}
+
+fn run_checks_on_manifest() -> anyhow::Result<()> {
+    let output = Command::new("../../../target/debug/cargo-semver-checks")
+        .current_dir("crate_current")
+        .args([
+            "semver-checks",
+            "check-release",
+            "--baseline-root=../crate_baseline/Cargo.toml",
+        ])
+        .output()?;
+    std::io::stdout().write_all(&output.stdout)?;
+    std::io::stdout().write_all(&output.stderr)?;
+    println!();
+    Ok(())
+}
+
+fn main() -> anyhow::Result<()> {
+    let client = crates_io_api::SyncClient::new(
+        "crates_io_api: cargo-semver-checks (obi1kenobi82@gmail.com)",
+        Duration::from_millis(10),
+    )?;
+
+    let mut query = CratesQuery::builder()
+        .page_size(100)
+        .sort(Sort::Downloads)
+        .build();
+    query.set_page(1);
+
+    let crate_page = client.crates(query)?;
+
+    for crate_info in crate_page.crates.into_iter() {
+        let versions = client.get_crate(&crate_info.name)?.versions;
+        for (i, version_i0) in versions.iter().enumerate() {
+            if i + 1 < versions.len() {
+                let version_i1 = &versions[i + 1];
+                if version_i0.yanked && !version_i1.yanked { // it gives more interesting results for testing
```

Comment:
This would certainly give interesting results, but aren't we primarily interested in checking the most recent version vs its baseline? I'd assume we want to make sure:
- new crates can adopt cargo-semver-checks without running into any errors
- we don't keep re-checking mostly-the-same code over and over again, and instead make better use of our limited CPU time

I think this is fine to have as a mode behind a flag, but I think the primary mode of this tool would just be to check the most recent "normal" version of each crate against its baseline.

Reply:
Removed yanked releases in 4f463225ca84523b8d04f140ce3ffec64a8524e8.
It nerveless didn't work, because cargo doesn't allow setting a yanked release as a dependency.

Reply:
I plan to add `--all-releases` and `--only-latest-release` as commands when the script will work.